catalog/transformer/catalog-transformer-service-atom/src/main/java/ddf/catalog/transformer/response/query/atom/AtomTransformer.java

```diff
@@ -233,6 +235,35 @@ public BinaryContent transform(SourceResponse sourceResponse, Map<String, Serial
       startIndex.setText(Integer.toString(sourceResponse.getRequest().getQuery().getStartIndex()));
     }
 
+    if (getCount(sourceResponse) != 0) {
```

Comment:
❗️ I know the method already had a very high cyclomatic complexity but we are making it worst now. We should take the time and simplify this code while we are here to reduce its complexity.

Reply:
I agree we should take the time to refactor this to reduce complexity, but it would be ideal to do it in a separate ticket to prevent scope creep.  These additions don't seem to make the class unmanageable or testing impractical.

Reply:
+1 to Ben, seems like this already took a step towards reducing complexity by separating out the addResultsElements.  This could be better, but incremental changes as it is updated seems reasonable for this rarely used capability.  

I did not get a complexity count, but I am less worried about it since it hasn't been flagged by any of the automated code analysis tools.

Reply:
I would tend to agree that a separate ticket would prevent scope creep, but for the record, both this method and `addResultElements` did actually get flagged by Codacy

[transform](https://www.codacy.com/app/Codice/ddf/file/12919297521/issues/source?bid=4642206&fileBranchId=5986171#l143)

[addResultElements](https://www.codacy.com/app/Codice/ddf/file/12919297521/issues/source?bid=4642206&fileBranchId=5986171#l258)

Reply:
I think the majority of this existed before the proposed changes.

Reply:
To clarify, the contents of `addResultElements()` were previously part of `transform()`. It looks like the code in `addResultElements()` is being added, but it's only being broken out into a new method. The code itself is unmodified except for the try-catch-finally blocks that @paouelle pointed out. Since Codacy is flagging those two methods now, I assume it was also flagging the previous version of `transform()` - which was much longer - as well.

Reply:
I do not think we should delay this. The fact that the other method was long before doesn't mean we shouldn't do it right now.

Reply:
I'm no expert on Codacy, but it looks like the complexity is reduced in this PR : https://www.codacy.com/app/Codice/ddf/pullRequest?prid=1134653.

Reply:
Since this PR does not introduce any more complexity to the AtomTransformer, the following ticket was created to address complexity concerns.
https://codice.atlassian.net/browse/DDF-3528